.github/workflows/build-envoy-image-ci.yaml

```diff
@@ -92,10 +107,11 @@ jobs:
           docker rm -fv cilium-envoy
           envoy_version=$(./cilium-envoy --version)
           echo ${envoy_version}
-          [[ "${envoy_version}" == *"${{ github.event.pull_request.head.sha }}"* ]]
+          [[ "${envoy_version}" == *"${{ github.event.pull_request.head.sha }}/${{ env.ENVOY_MINOR_RELEASE }}"* ]]
```

Comment:
Locally I get this output for `./cilium-envoy --version`:
```
./cilium-envoy  version: b308011020900e90579ad5927857d01f5de8016f/1.24.2/Distribution/DEBUG/BoringSSL
```
Note that there is no `v` in `1.24.2`! So the change here would fail the workflow for non-matching version.